src/librustc/ty/context.rs

```diff
@@ -2810,6 +2814,11 @@ pub fn provide(providers: &mut ty::query::Providers) {
         assert_eq!(id, LOCAL_CRATE);
         tcx.crate_name
     };
+    providers.get_lib_features = |tcx, id| {
+        assert_eq!(id, LOCAL_CRATE);
+        // FIXME(#42293): see comment below.
+        tcx.dep_graph.with_ignore(|| Lrc::new(middle::lib_features::collect(tcx)))
```

Comment:
Actually, this should work just fine without the `with_ignore`. That comment below is really old.